File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/common/config/WebConfig.java
package com.multicampus.gamesungcoding.a11ymarketserver.common.config;

import org.springframework.context.annotation.Configuration;
import org.springframework.http.MediaType;
import org.springframework.http.converter.HttpMessageConverter;
import org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import java.util.ArrayList;
import java.util.List;

@Configuration
public class WebConfig implements WebMvcConfigurer {

    @Override
    public void extendMessageConverters(List<HttpMessageConverter<?>> converters) {
        for (var converter : converters) {
            if (converter instanceof AbstractJackson2HttpMessageConverter jacksonConverter) {
                var supportedMediaTypes = new ArrayList<>(jacksonConverter.getSupportedMediaTypes());
                
                supportedMediaTypes.add(MediaType.APPLICATION_OCTET_STREAM);

                jacksonConverter.setSupportedMediaTypes(supportedMediaTypes);
            }
        }
    }
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/common/config/WebConfig.kt
package com.multicampus.gamesungcoding.a11ymarketserver.common.config

import org.springframework.context.annotation.Configuration
import org.springframework.http.MediaType
import org.springframework.http.converter.HttpMessageConverter
import org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebConfig : WebMvcConfigurer {
    override fun extendMessageConverters(converters: MutableList<HttpMessageConverter<*>>) {
        converters.filterIsInstance<AbstractJackson2HttpMessageConverter>().forEach { converter ->
            converter.supportedMediaTypes = converter.supportedMediaTypes + MediaType.APPLICATION_OCTET_STREAM
        }
    }
}
